src/client.rs

```diff
@@ -586,6 +580,10 @@ where
             self.application_name.clone(),
         );
 
+        // Internal buffer, where we place messages until we have to flush
+        // them to the backend.
+        let mut message_buffer = BytesMut::with_capacity(8196);
```

Comment:
Our current buffer is allocated once. This creates a `malloc(8196)` call for each transaction which will be much slower. Traditionally, network applications allocate one buffer on startup and re-use it.